### DISCUSSION TITLE
MyClass make_from_json(const json& j) for Objects without default/move constructor

### DISCUSSION THREAD

**Discussion by mqnc** [nlohmann/json#3386] [Ideas]
Because I want to initialize objects with constant members from json, it would be nice if there was such a mechanism. I think it could work somewhat like this:

```cpp
// in json.hpp:
template<typename T>
T make_from_json(const nlohmann::json&);
...
detail::enable_if_t <detail::has_make_from_json> ... ValueType get_impl(...){
    return make_from_json(*this);
```

and then it can be used like this:

```cpp
template<>
MyClass make_from_json(const json& j){...}

auto myObject = myJson.get<MyClass>();
```

Would that be possible or does it clash with some other mechanism or policy or is generally a bad idea for some reason?

The thing is, I have been creating and using make_from_json for some custom types already but without the get specialization in the library, I can't combine that with standard containers, for instance creating std::vector<MyClass> from a json object. Or I have to implement make_from_json for all the container types I use.

**Comment by nlohmann** [comment]
Can https://json.nlohmann.me/features/arbitrary_types/#how-can-i-use-get-for-non-default-constructiblenon-copyable-types help you?

**Reply by mqnc** [reply]
It does! Thank you! I saw this before but I thought a type with const members is not move constructible but I think I confused that.
